app/cdap/components/WranglerGrid/SelectColumnPanel/ColumnsList/index.tsx

```diff
@@ -0,0 +1,218 @@
+import { ISelectColumnsListProps } from 'components/WranglerGrid/SelectColumnPanel/ColumnsList/types';
+import { IHeaderNamesList } from 'components/WranglerGrid/SelectColumnPanel/types';
+import ColumnTable from 'components/WranglerGrid/SelectColumnPanel/DataTable';
+import { MULTI_SELECTION_COLUMN } from 'components/WranglerGrid/SelectColumnPanel/constants';
+import SelectedColumnCountWidget from 'components/WranglerGrid/SelectColumnPanel/CountWidget';
+import { IMultipleSelectedFunctionDetail } from 'components/WranglerGrid/SelectColumnPanel/types';
+import { SELECT_COLUMN_LIST_PREFIX } from 'components/WranglerGrid/SelectColumnPanel/constants';
+import { NormalFont, SubHeadBoldFont } from 'components/common/TypographyText';
+import { Box, IconButton } from '@material-ui/core';
+import SearchIcon from '@material-ui/icons/Search';
+import styled from 'styled-components';
+import { grey } from '@material-ui/core/colors';
+import {
+  getColumnsSupportedType,
+  getFilteredColumn,
+} from 'components/WranglerGrid/SelectColumnPanel/utils';
+
+const SearchIconComponent = styled(SearchIcon)`
+  &.MuiSvgIcon-root {
+    width: 24px;
+    height: 24px;
+  }
+`;
+
+const SelectColumnSearchInput = styled.input`
+  margin-right: 5px;
+  border: none;
+  border-bottom: 1px solid transparent;
+  margin-bottom: 5px;
+  &:focus {
+    border-bottom: 1px solid ${grey[700]};
+    outline: none;
+  }
+`;
+
+const SelectColumnWrapper = styled(Box)`
+  height: 90%;
+`;
+
+const SelectColumnInnerWrapper = styled(Box)`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-top: 10px;
+`;
+
+const FlexWrapper = styled(Box)`
+  display: flex;
+  height: 100%;
+  align-items: center;
+`;
+
+const CenterAlignBox = styled(Box)`
+  text-align: center;
+`;
+
+const SelectColumnSearchBox = styled(Box)`
+  position: relative;
+  display: flex;
+`;
+
+const SearchIconButton = styled(IconButton)`
+  padding: 5px 0px 5px 5px;
+
+  &.MuiIconButton-root:hover {
+    background-color: transparent;
+  }
+  & .MuiTouchRipple-root {
+    display: none;
+  }
+`;
+
+export default function({
+  transformationDataType,
+  selectedColumnsCount,
+  columnsList,
+  setSelectedColumns,
+  dataQuality,
+  transformationName,
+  selectedColumns,
+}: ISelectColumnsListProps) {
+  const [columns, setColumns] = useState<IHeaderNamesList[]>(columnsList);
+  const [isSingleSelection, setIsSingleSelection] = useState<boolean>(true);
+  const ref = useRef(null);
+
+  const columnsAsPerType = getColumnsSupportedType(transformationDataType, columnsList);
+  const filteredColumnsOnType = getFilteredColumn(transformationDataType, columnsList);
+
+  useEffect(() => {
+    const multiSelect: IMultipleSelectedFunctionDetail[] = MULTI_SELECTION_COLUMN?.filter(
+      (functionDetail: IMultipleSelectedFunctionDetail) =>
+        functionDetail.value === transformationName
+    );
+
+    if (multiSelect.length) {
+      setIsSingleSelection(false);
```

Comment:
as this is executed on initial rendering of the component
I gues this one is same as the state

why not add the condition directly in the state while initializing the state variable
maybe !!multiSelect.length should do
const [isSingleSelection, setIsSingleSelection] = useState<boolean>(!!multiSelect.length);

plz check if this works or find an alternative for this if the thought seems right
plz let me know if otherwise with proper explanation